drivers/firmware/scmi/reboot.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_ARM_SCMI_REBOOT)
+
+#include <zephyr/drivers/firmware/scmi/system.h>
+#include <string.h>
+
+#if defined(CONFIG_REBOOT)
```

Comment:
`config ARM_SCMI_REBOOT` `depends on ARM_SCMI_SYSTEM && REBOOT`

Thus, this #if is always selected. I believe it should be safe to remove these conditions